pkg/catalog/proxy.go

```diff
@@ -17,14 +19,20 @@ func (mc *MeshCatalog) ExpectProxy(cn certificate.CommonName) {
 }
 
 // RegisterProxy implements MeshCatalog and registers a newly connected proxy.
-func (mc *MeshCatalog) RegisterProxy(p *envoy.Proxy) {
+func (mc *MeshCatalog) RegisterProxy(proxy *envoy.Proxy) {
 	mc.connectedProxiesLock.Lock()
```

Comment:
you could very well turn this lock + map into a sync.map at this stage

Reply:
Good idea - let me create another PR for that - in a few minutes!

Reply:
@eduser25 Here is the PR changing the `connectedProxies` to a `sync.Map`:  https://github.com/openservicemesh/osm/pull/2061  (another 2 PRs coming)